.github/workflows/npm_publish.yml

```
@@ -0,0 +1,30 @@
+name: npm publish
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install Depencencies
+        run: yarn install
+
+      - name: Build
+        run: yarn build:all
+  
+  publish:
+      - name: Publish npm package
```

If you'd like the `build` to pass before doing a publish, you can have the publish job wait on the build job with:
```suggestion
  publish:
    needs: ["build"]
      - name: Publish npm package
```